### PR TITLE
Sort the internal list of voters by priority

### PR DESCRIPTION
**Pull request by FlashBlack**
And piece of code in knp-menubundle
see: https://github.com/KnpLabs/KnpMenuBundle/pull/268

### CODE REVIEW

File: src/Knp/Menu/Matcher/Matcher.php [hunk 1]
```diff
@@ -13,9 +13,14 @@ class Matcher implements MatcherInterface
     private $cache;
 
     /**
+     * @var array[]
+     */
+    private $voters = [];
```
Comment (stof) [inline src/Knp/Menu/Matcher/Matcher.php]: must keep compatibility with PHP 5.3

**Comment by stof** [comment]
Does KnpMenu really need a concept of priority ? I think running voters in the order in which they are added is enough here. the concept of priority should be used by the code configuring the matcher if needed (it may not be needed depending of how the configuration is done).
KnpMenuBundle could then use the priority to sort tagged services (to give control over the order in which voters are added). Symfony does that in many places (the EventDispatcher is one of the only place where the priority needs to be known by the code rather than the configuration layer btw)

**Comment by stof** [comment]
so I'm -1 on this change in the library.

**Comment by akovalyov** [comment]
@FlashBlack could you add a little bit of details why do you need this change?

**Comment by Nek-** [comment]
@akovalyov notice that the implementation is wrong, and need a test.

Btw I :-1: too as voter don't need to be prioritized: if a voter say "no", then it's no. No matter the order. The only point here is authorize bad application design (things that the voter should not do in the voter).

**Comment by stof** [comment]
If you need a strict order for voters, just register them in the right order. I don't want to add the concept of priority in the library.

**Pull request closed without merging**